src/main/java/com/dougnoel/sentinel/steps/TableSteps.java

```diff
@@ -145,4 +147,25 @@ public static void storeTableValues(int pageNumber, String tableName) {
     public static void clickColumnHeaderToSort(String columnName, String tableName) {
     	getElementAsTable(tableName).clickColumnHeader(columnName);
     }
+
+	/**
+	 * Clicks the xpath of element in row which have stored value in column name
+	 * <p>
+	 * <b>Gherkin Examples:</b>
+	 * <ul>
+	 * <li>I find the Name column in the Project Files Table in row with value entered for the Search Input and click the xpath //i[contains(concat(' ', @class, ' '), ' fa-toggle-on ')]</li>
+	 * </ul>
+	 * @param columnName String the name of the column to click the header of
+	 * @param tableName String the name of the table element
+	 * @param key String the key used to retrieve the value
+	 * @param xpath String xpath value to element to click
+	 */
+	@When("^I find the (.*?) column in the (.*?) in row with value (?:entered|selected|used) for the (.*?) and click the xpath (.*?)$")
+	public static void clickStoredTextRowContainsXpath(String columnName, String tableName, String key, String xpath) {
+		var textToMatch = Configuration.toString(key);
+		String errorMessage = SentinelStringUtils.format("No previously stored text was found for the \"{}\" key.", key);
+		Assert.assertNotNull(errorMessage, textToMatch);
+		var rowIndex = getElementAsTable(tableName).getAllCellDataForColumn(columnName).indexOf(textToMatch);
+		getElementAsTable(tableName).getElementInRowThatContains(rowIndex, By.xpath(xpath)).click();
+	}
```

Comment:
The null assert should be handled as a try/catch. Otherwise we are doing an evaluation every time which slows down the program.